### PR TITLE
Combine shallow and retained size columns into one to save horizontal space.

### PR DESCRIPTION
**Pull request by polina-c**
RELEASE_NOTE_EXCEPTION=[too small to mention]

https://user-images.githubusercontent.com/12115586/220806868-3aa3cc42-185b-4140-b705-fab1ad5d8ea8.mov

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart [hunk 1]
```diff
@@ -243,34 +239,40 @@ class _SizeColumn extends ColumnData<DiffClassStats> {
...
-    _SizeColumn(_DataPart.created, _SizeType.shallow),
-    _SizeColumn(_DataPart.deleted, _SizeType.shallow),
-    _SizeColumn(_DataPart.delta, _SizeType.shallow),
-    _SizeColumn(_DataPart.persisted, _SizeType.shallow),
-    _SizeColumn(_DataPart.created, _SizeType.retained),
-    _SizeColumn(_DataPart.deleted, _SizeType.retained),
-    retainedSizeDeltaColumn,
-    _SizeColumn(_DataPart.persisted, _SizeType.retained),
+    if (sizeTypeToShow == SizeType.shallow) ...[
+      _SizeColumn(_DataPart.created, SizeType.shallow),
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:265]: Can we remove the spreads and instead just do this?

```dart
late final List<ColumnData<DiffClassStats>> columnList =
    <ColumnData<DiffClassStats>>[
  // ...
  _SizeColumn(_DataPart.created, sizeTypeToShow),
  _SizeColumn(_DataPart.deleted, sizeTypeToShow),
  sizeDeltaColumn,
  _SizeColumn(_DataPart.persisted, sizeTypeToShow),
];
```

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart [hunk 2]
```diff
@@ -282,60 +284,99 @@ class ClassesTableDiff extends StatelessWidget {
...
+      ColumnGroup.fromText(
+        title: '',
+        range: const Range(0, 1),
+      ),
+      ColumnGroup.fromText(
+        title: 'Instances',
+        range: const Range(1, 5),
+        tooltip: nonGcableInstancesColumnTooltip,
+      ),
+      if (sizeType == SizeType.shallow)
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:335]: Can we remove these conditions and just use a single `ColumnGroup` for `sizeTitle`?

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart [hunk 3]
```diff
@@ -282,60 +284,99 @@ class ClassesTableDiff extends StatelessWidget {
...
-      selectionNotifier: selection,
-      onItemSelected: (_) => ga.select(
-        gac.memory,
-        gac.MemoryEvent.diffClassDiffSelect,
-      ),
-      defaultSortColumn: columns.retainedSizeDeltaColumn,
-      defaultSortDirection: SortDirection.descending,
+    return ValueListenableBuilder<SizeType>(
+      valueListenable: sizeTypeToShowForDiff,
+      builder: (context, sizeType, ___) {
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:354]: Ubernit: `s/___/_/`?

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart [hunk 4]
```diff
@@ -282,60 +284,99 @@ class ClassesTableDiff extends StatelessWidget {
...
   }) : super(key: key);
 
   final List<DiffClassStats> classes;
   final ValueNotifier<DiffClassStats?> selection;
   final AdaptedHeapData before;
   final AdaptedHeapData after;
+  final ValueNotifier<SizeType> sizeTypeToShowForDiff;
+
+  List<ColumnGroup> _columnGroups(SizeType sizeType, BuildContext context) {
+    final sizeTitle = maybeWrapWithTooltip(
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:297]: This looks like it should be a Widget.

**Review by bkonyi — APPROVED** [review]

**Review by kenzieschmoll — COMMENTED** [review]
can we add a test for toggling between shallow and retained and verifying that the data content changes properly?

**Pull request merged**